app/src/templates/dashboard.html

```diff
@@ -119,39 +119,16 @@ <h4>Test Yourself!</h4>
                 <div class="card-body3">
                     <div id="leaderboard-card-header">
                         <h4>Weekly Leaderboard</h4>
-                        <!-- <a href="{{ leaderboard_url }}" id="leaderboard-link">
-                            
-                        </a> -->
-                        <!-- put the button line inside the <a> tag above after we set up the dashboard route to send in 
-                            the leaderboard info -->
-                        <button id="view-leaderboard">VIEW ALL</button>
+                        <button id="view-leaderboard" onclick="window.location.href='/leaderboard'">VIEW ALL</button>
                     </div>
                     <ul id="leaderboard-list">
+                        {% for user in leaderboard_data %}
                         <li class="leaderboard-entry">
-                            <span class="leaderboard-rank">1</span>
-                            <span class="leaderboard-name">James Smith</span>
-                            <span class="leaderboard-points">100 XP</span>
-                        </li>
-                        <li class="leaderboard-entry">
-                            <span class="leaderboard-rank">2</span>
-                            <span class="leaderboard-name">John Smith</span>
-                            <span class="leaderboard-points">90 XP</span>
-                        </li>
-                        <li class="leaderboard-entry">
-                            <span class="leaderboard-rank">3</span>
-                            <span class="leaderboard-name">Jane Smith</span>
-                            <span class="leaderboard-points">80 XP</span>
-                        </li>
-                        <li class="leaderboard-entry">
-                            <span class="leaderboard-rank">4</span>
-                            <span class="leaderboard-name">Kevin Smith</span>
-                            <span class="leaderboard-points">70 XP</span>
-                        </li>
-                        <li class="leaderboard-entry">
-                            <span class="leaderboard-rank">5</span>
-                            <span class="leaderboard-name">Mike Smith</span>
-                            <span class="leaderboard-points">60 XP</span>
+                            <span class="leaderboard-rank">{{ loop.index }}</span>
+                            <span class="leaderboard-name">{{ user.username }}</span>
+                            <span class="leaderboard-points">{{ user.points }} XP</span>
```

Comment:
Switch to user.points.points to correctly access the user points value from the points model.